src/Mantasflowers.Services/DataAccess/Repositories/UserOrderRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Mantasflowers.Domain.Entities;
+using Mantasflowers.Persistence;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.Services.DataAccess.Repositories
+{
+    public class UserOrderRepository : BaseRepository<UserOrder>, IUserOrderRepository
+    {
+        public UserOrderRepository(DatabaseContext dbContext)
+            : base(dbContext) { }
+
+        public async override Task<UserOrder> CreateAsync(UserOrder entity)
+        {
+            _dbContext.Entry(entity.Order).State = EntityState.Unchanged;
+
+            entity.User = await _dbContext.Users.FindAsync(entity.UserId);
+            _dbContext.Entry(entity.User).State = EntityState.Unchanged;
```

Comment:
Besides the hacky looking implementation, why is order being taken as is, but user is being fetched anew?

Reply:
So I wouldn't have to fetch order again, since it is created and saved beforehand in a payment transaction. EF Core is supposed to manage states by itself, however, there are limits (as are in this case).

Reply:
Could it be that it doesn't work for you because of Automapper?

Reply:
There is no mapper used in between these specific operations.

Reply:
I think I saw entity, the parameter (or part of it) being mapped from request object somewhere. Maybe I'm wrong.

Reply:
```cs
order = await _orderService.CreateOrderAsync(request.Order);
await _unitOfWork.SaveChangesAsync();

if (userId.HasValue)
{
    await _orderService.LinkUserToOrderAsync(userId.Value, order);
}
```
After this section, it goes straight to save UserOrder entity.